spec/engine/matchers/depencdency_matcher_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+
+describe SplitIoClient::DependencyMatcher do
+  let(:split_treatment) { double }
+
+  it 'matches' do
+    allow(split_treatment).to receive(:call).with({ matching_key: 'foo' }, 'foo', nil).and_return(treatment: 'yes')
+
+    expect(described_class.new('foo', %w(on yes true)).match?('foo', split_treatment, nil)).to eq(true)
+  end
+
+  it 'does not match' do
+    allow(split_treatment).to receive(:call).with({ matching_key: 'foo' }, 'foo', nil).and_return(treatment: 'no')
+
+    expect(described_class.new('foo', %w(on yes true)).match?('foo', split_treatment, nil)).to eq(false)
+  end
+
+  xit 'caches values' do
```

Comment:
Deleted it. I was thinking of putting cache test there, but ended up with `split_treatment_spec.rb`